lib/client.js

```diff
@@ -89,9 +89,18 @@ function Client (broker, conn, req) {
   conn.on('end', this.close.bind(this))
   this._eos = eos(this.conn, this.close.bind(this))
 
-  this.deliver0 = function deliverQoS0 (_packet, cb) {
+  const getToForwardPacket = (_packet) => {
+    if (_packet.clientId === that.id && _packet.nl) return
```

Comment:
Could you add a reference to MQTTv5 Section here? 
```suggestion
    // Mqttv5 3.8.3.1: https://docs.oasis-open.org/mqtt/mqtt/v5.0/mqtt-v5.0.html#_Toc3901169
    // prevent to forward messages sent by the same client when no-local flag is set
    if (_packet.clientId === that.id && _packet.nl) return
```